app/api/discount_codes.py

```diff
@@ -261,47 +243,39 @@ def before_get_object(self, view_kwargs):
                                       "Please verify your permission. You must be admin to view event\
                                       discount code details")
 
-    def before_update_object(self, obj, data, view_kwargs):
+    def before_update_object(self, discount, data, view_kwargs):
         """
         Method to edit object
+        :param discount:
         :param data:
         :param view_kwargs:
         :return:
         """
-        discount = self.session.query(DiscountCode).filter_by(id=view_kwargs.get('id')).one()
-        if not discount:
-            raise ObjectNotFound({'parameter': '{id}'}, "DiscountCode:  not found")
-
         if 'used_for' in data:
             used_for = data['used_for']
-        elif not 'used_for' in data:
+        else:
             used_for = discount.used_for
 
-        if discount.used_for == 'ticket' and (current_user.is_coorganizer(view_kwargs.get('event_id'))
-           or current_user.is_organizer(view_kwargs.get('event_id') or current_user.is_admin)) and used_for != 'event':
+        if discount.used_for == 'ticket' and has_access('is_coorganizer', event_id=view_kwargs.get('event_id'))\
+           and used_for != 'event':
             self.schema = DiscountCodeSchemaTicket
 
-        elif discount.used_for == 'event' and current_user.is_admin and used_for != 'ticket':
+        elif discount.used_for == 'event' and has_access('is_admin') and used_for != "ticket":
```

Comment:
Actually not changed 😅 maybe I wrote it again so added single quotes. should I change them to double?